src/components/file-browser/FilesInDirectory.ts

```diff
@@ -64,6 +65,15 @@ const FilesInDirectory = (props: FilesInDirectoryProps) => {
   const { uploadState, uploadFiles, cancelUpload } = useUploader((file) => {
     return provider.uploadFileToDirectory(path, file);
   });
+  useOnUploadFinished(uploadState, (finishedUploadState) => {
+    finishedUploadState.errors.forEach(({ file, error }) => {
+      notify('error', 'Error uploading file', {
+        id: 'file-browser-upload-error',
+        message: `Failed to upload ${file.name}`,
+        detail: error,
+      });
+    });
+  });
```

Comment:
in a corner case where the user tried to upload 500 files and they all failed (maybe a permissions issue) … will the toast notification show "1/500"? Might be good to put a limit on it, if there are more than 5 or 10 errors or something, show only one error with "${n} files failed to upload."

Reply:
I'm not convinced this is an issue... the current behavior lets the user see how many errors occurred and the details of each one (if they want). But it still behaves as one error notification: they can dismiss it with one click, they don't have to dismiss each error individually.

I'm inclined to merge this as an improvement and revisit later if UX wants to change it.